src/azul/service/responseobjects/collection_data_access.py

```diff
@@ -0,0 +1,151 @@
+import requests
+from azul import config
+
+logger = getLogger(__name__)
+
+
+class CollectionDataAccess:
+    DEFAULT_MAX_BACKOFF_TIME = 60
+
+    def __init__(self, access_token:str):
+        self.access_token = access_token
+
+    def get(self, uuid:str, version:str):
+        url = self.endpoint_url('collections', uuid)
+        query = {
+            "version": version,
+            "replica": "aws"
+        }
+        response = self.send_request(uuid, 'get', url, query,
+                                     exception_class=RetrievalError)
+        collection = response.json()
+        return dict(uuid=uuid, version=version, collection=collection)
+
+    def create(self, uuid:str, name:str, description:str, version:str, items:List[Dict[str, str]]):
+        url = self.endpoint_url('collections')
+        query = {
+            "uuid": uuid,
+            "version": version,
+            "replica": "aws"
+        }
+        payload = {
+            "name": name,
+            "description": description,
+            "details": {},
+            "contents": items
+        }
+        response = self.send_request(uuid, 'put', url, query, payload,
+                                     expected_status_code=201,
+                                     exception_class=CreationError)
+        collection = response.json()
+        return dict(uuid=collection['uuid'], version=collection['version'])
+
+    def append(self, uuid:str, version:str, items:List[Dict[str, str]]):
+        url = self.endpoint_url('collections', uuid)
+        query = {
+            "version": version,
+            "replica": "aws"
+        }
+        payload = {
+            "add_contents": [item for item in items]
+        }
+        response = self.send_request(uuid, 'patch', url, query, payload,
+                                     exception_class=UpdateError)
+        collection = response.json()
+        return dict(uuid=collection['uuid'], version=collection['version'])
+
+    def send_request(self, uuid, method:str, url:str, params, payload:Optional[Any]=None, delay:Optional[int]=None,
+                     request_id:Optional[str]=None, expected_status_code:Optional[int]=200, exception_class=None):
+        request_id = request_id or str(uuid4())
+        # delay_factor is for automatic retry with exponential backoff.
+        if delay is not None:
+            if delay > self.DEFAULT_MAX_BACKOFF_TIME:
+                logger.warning('Request %s: The request fails to respond within the time limit.', request_id)
+                raise ServerTimeoutError(uuid)
+            logger.info('Request %s: Retrying in %d s', request_id, delay)
+            sleep(delay)
+            logger.info('Request %s: Resuming', request_id)
+        request_params = dict(params=params,
+                              headers=dict(Authorization=self.access_token))
+        if payload:
+            request_params['json'] = payload
+        # FIXME "requests" will be replaced with "DSSClient".
```

Comment:
Didn't we agree we would do this? I can't remember.

Reply:
In our last call, we agree not to use `DSSClient` for this PR.